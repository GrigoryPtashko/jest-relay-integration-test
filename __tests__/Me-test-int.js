import React from 'react'; // eslint-disable-line import/imports-first
import Relay from 'react-relay'; // eslint-disable-line import/imports-first
import renderer from 'react-test-renderer';
import RelayStore from '../src/RelayStore';
import { createRenderer } from '../src/RelayUtils';

RelayStore.reset(new Relay.DefaultNetworkLayer('http://hashnews.cfapps.io/q'));

const delay = value => new Promise(resolve => setTimeout(() => resolve(), value));

const RootQuery = {
  root: () => Relay.QL`
    query {
      root
    }
  `,
};

class AppRoot extends React.Component {
  static propTypes = {
    root: React.PropTypes.object.isRequired,
  };

  render() {
    return (
      <div>
        {this.props.root.me.authorities[0].authority}
      </div>
    );
  }
}

it('can make request to /q anyway', async () => {
  const AppCnt = createRenderer(AppRoot, {
    queries: RootQuery,
    fragments: {
      root: () => Relay.QL`
        fragment on Root {
          me {
            firstName
            email
            authorities {
              authority
            }
          }
        }
      `,
    },
  });

  const tree = renderer.create(<AppCnt />);

  await delay(3000);

  expect(tree.toJSON()).toMatchSnapshot();
});

