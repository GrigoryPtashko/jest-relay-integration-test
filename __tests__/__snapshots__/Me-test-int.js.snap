exports[`test can make request to /q anyway 1`] = `
<div>
  ROLE_ANONYMOUS
</div>
`;
